docs/user_manual/components.md

```diff
@@ -695,15 +695,116 @@ A `fault` has no steady state output.
-| `FaultType.single_phase_to_ground` | `FaultPhase.a`   | One phase is grounded with fault impedance, and other phases are open.  |
-| `FaultType.two_phase`              | `FaultPhase.bc`  | Two phases are connected with fault impedance.                          |
-| `FaultType.two_phase_to_ground`    | `FaultPhase.bc`  | Two phases are connected with fault impedance then grounded.            |
+| `fault_type`                       | `fault_phase`    | description                                                            |
+| ---------------------------------- | ---------------- | ---------------------------------------------------------------------- |
+| `FaultType.three_phase`            | `FaultPhase.abc` | Three phases are connected with fault impedance.                       |
+| `FaultType.single_phase_to_ground` | `FaultPhase.a`   | One phase is grounded with fault impedance, and other phases are open. |
+| `FaultType.two_phase`              | `FaultPhase.bc`  | Two phases are connected with fault impedance.                         |
+| `FaultType.two_phase_to_ground`    | `FaultPhase.bc`  | Two phases are connected with fault impedance then grounded.           |
 
 In case the `fault_phase` is not specified or is equal to `FaultPhase.default_value`, the power-grid-model assumes the following fault phases for different values of `fault_type`.
 
+## Regulator
+
+* type name: `regulator`
+* base: {hoverxreftooltip}`user_manual/components:base`
+
+`regulator` is an abstract regulator that is coupled to a given `regulated_object`. For each `regulator`, a switch is defined between
+the `regulator` and the `regulated_object`. Which object types are supported as `regulated_object` is regulator type-dependent.
+
+#### Input
+
+| name               | data type | unit | description                               | required |  update  |        valid values         |
+| ------------------ | --------- | ---- | ----------------------------------------- | :------: | :------: | :-------------------------: |
+| `regulated_object` | `int32_t` | -    | ID of the regulated object                | &#10004; | &#10060; | a valid regulated object ID |
+| `status`           | `int8_t`  | -    | connection status to the regulated object | &#10004; | &#10004; |         `0` or `1`          |
+
+### Transformer tap regulator
+
+* type name: `transformer_tap_regulator`
+* base: {hoverxreftooltip}`user_manual/components:regulator`
 
+`transformer_tap_regulator` defines a regulator for transformers in the grid.
+At the time of writing, a transformer tap regulator regulates a component that is either a {hoverxreftooltip}`user_manual/components:transformer` or a {hoverxreftooltip}`user_manual/components:Three-Winding Transformer`.
+
+The transformer tap regulator overloads the `tap_pos` of the transformer it regulates in the range set by the user via `tap_min` and `tap_max` (i.e., `(tap_min <= tap_pos <= tap_max)` or `(tap_min >= tap_pos >= tap_max)`).
```

Comment:
While it is most accurate, `overloads` here is a non electrical term users can make do without.
(This is a nitpick, resolve without changes if you wish to)